### PR TITLE
[W4.4b][W09-A1]Yu Hong Herald

### PR DESCRIPTION
**Pull request by yuhongherald**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street, -, -";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address must be block, street, unit and postalcode. " +
+            "Seeparate with commas. Use - to indicate blank field.";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final int fields = 4;
+
+    public final Block block;
```
Comment (eugenepeh) [inline src/seedu/addressbook/data/person/Address.java:23]: public value doesn't seems to be used anywhere else

File: src/seedu/addressbook/data/person/address/Block.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package seedu.addressbook.data.person.address;
+
+public class Block
```
Comment (eugenepeh) [inline src/seedu/addressbook/data/person/address/Block.java:3]: missing header comments for classes

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -26,7 +39,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] values = trimmedAddress.split(",");
         this.value = trimmedAddress;
+        if (values.length != 4) {
```
Comment (eugenepeh) [inline src/seedu/addressbook/data/person/Address.java:44]: take note of the use of magic numbers, you've declared the `fields` number above but its not used.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street, -, -";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address must be block, street, unit and postalcode. " +
+            "Seeparate with commas. Use - to indicate blank field.";
```
Comment (eugenepeh) [inline src/seedu/addressbook/data/person/Address.java:19]: typo in message

**Review by eugenepeh — COMMENTED** [review]
Good job on completing the activity.
Some comments added. Please close the PR after reading the comments.

File: src/seedu/addressbook/data/person/Address.java (hunk 3 above)
Comment (yuhongherald) [reply src/seedu/addressbook/data/person/Address.java:44]: oh haha sorry i forgot

**Pull request closed without merging**